test/testutil.c

```diff
@@ -145,19 +145,110 @@ int run_tests(const char *test_prog_name)
     return EXIT_SUCCESS;
 }
 
+void fail_message(const char *desc, va_list args, const char *type,
```

Comment:
Ideally I'd like to see `__FILE__` and `__LINE__` here. You'd need to move the implementation to a header, though.

With `__FILE__` and `__LINE__` we probably won't even need `desc` in most cases, so we could make custom descriptions optional (less typing):

`ret = TEST_int_eq(a, b);`

And optionally

```
ret = TEST_int_eq_msg(a[i], b[i], "%d th element mismatch", i);
```